Mapping Tools/Classes/SystemTools/ListenerManager.cs

```diff
@@ -104,6 +104,52 @@ public class ListenerManager {
             EditorReaderStuff.CoolSave();
         }
 
+        private void SmartQuickRun() {
+            if (!SettingsManager.Settings.SmartQuickRunEnabled) { QuickRunCurrentTool(); return; }
+
+            if (EditorReaderStuff.TryGetFullEditorReader(out EditorReader reader)) {
+                EditorReaderStuff.GetNewestVersion(IOHelper.GetCurrentBeatmap(), out List<BeatmapHelper.HitObject> selected, reader);
```

Comment:
can you do this without GetNewestVersion? This method adds more steps than probably necessary. The EditorReader object also contains all your selected objects and some more stuffs.